src/loaders/TiffLoader.js

```diff
@@ -0,0 +1,252 @@
+import { isInTileBounds } from './utils';
+
+/**
+ * This class serves as a wrapper for reading tiff data.
+ * @param {Object} args
+ * @param {Object} args.tiff geotiffjs tiff object.
+ * @param {Object} args.pool Pool that implements a `decode` function.
+ * @param {Object} args.firstImage First image (geotiff Image object) in the tiff (containing base-resolution data).
+ * @param {Object} args.dimensions Dimensions for the loader.
+ * @param {Object} args.metadata The metadata object
+ * @param {Array} args.offsets The offsets of each IFD in the tiff.
+ * @param {Boolean} args.isRgb Whether or not this tiff represents an rgb image.
+ * @param {Boolean} args.isInterleaved Whether or not this tiff represents an interleaved image.
+ * @param {String} args.dtype The dtype of the image like <u2 or <f4.
+ * @param {Object} args.physicalSizes An object like { x: { value, unit }, y: { value, unit } } where the first keys,
+ * x and y for example, are from the `dimensions and the value is the length and the unit is the units for the value
+ * to be interpreted in, like mm for millimeters.
+ * */
+export default class TiffLoader {
+  constructor({
+    tiff,
+    pool,
+    firstImage,
+    dimensions,
+    offsets,
+    metadata,
+    isRgb,
+    isInterleaved,
+    dtype,
+    physicalSizes
```

Comment:
Just a general comment. This is a huge constructor for a base class. 

For example, do all Tiffs have `physicalSizes`? It seems like logic that used to be in the OMETIFFLoader constructor has been moved to `createTiffLoader`.